frontend/src/App.js

```diff
@@ -16,6 +18,11 @@ export default function App() {
   })
 
   const [singleTrack, setSingleTrack] = useState({})
+  const [bookmarks, setBookmarks] = useState(getBookmarks || [])
+
+  useEffect(() => {
+    storeBookmarks(bookmarks)
+  }, [bookmarks])
 
```

Comment:
Does this end in an infinite loop, doesn't it?

Reply:
No, this is the local storage, which has no side effects.